libUtils/windowsChromeDriver.py

```diff
@@ -0,0 +1,91 @@
+import subprocess
+import requests
+import os
+import zipfile
+
+class BrowserManager:
+    def __init__(self,extract_dir=".//dependencies//chrome-driver"):
+        self.extract_dir = extract_dir
+    def get_chrome_version(self):   #    Get the Chrome browser version installed on Windows.
+        try:
+            output = subprocess.check_output(
+                r'reg query "HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon" /v version',
+                shell=True,
+                encoding='utf-8'
+            )
+            # Extract version from the command output
+            version = output.split()[-1]
+            print(f"Chrome Version: {version}")
+            return version
+
+        except Exception as e:
+            raise Exception(f"Error getting Chrome version: {e}")
+
+    def get_chrome_driver_download_url(self, chrome_version):
+        try:
+            if not chrome_version:
+                raise Exception("Chrome version is required to fetch the download URL.")
+
+            download_url = f"https://storage.googleapis.com/chrome-for-testing-public/{chrome_version}/win64/chromedriver-win64.zip"
+            print(f"Download URL: {download_url}")
+
+            response = requests.head(download_url)
+
+            if response.status_code != 200:
+                raise Exception(
+                    f"Failed to get ChromeDriver for Chrome {chrome_version}. HTTP Status Code: {response.status_code}")
+
+            return download_url
+        except Exception as e:
+            raise Exception(f"Error getting ChromeDriver download URL: {e}")
+
+    def download_and_extract_chromedriver(self,download_url):
+        """Download and extract the ChromeDriver zip file."""
+        try:
+            response = requests.get(download_url, stream=True)
```

Comment:
separate two differrent function ,   download_zip_file(url) and extract_zip_file()    , get_driver()